src/egl/drivers/dri2/platform_android_gralloc.cpp

```diff
@@ -0,0 +1,23 @@
+#include "egl_dri2.h"
+#include <cutils/log.h>
+extern "C" int resolve_gralloc1_functions(struct dri2_egl_display *dri2_dpy);
+
+int resolve_gralloc1_functions(struct dri2_egl_display *dri2_dpy){
```

Comment:
I preferred using C++ casting for classes (reinterpret_cast). I did not try using C here as we already faced issues in gralloc1.0 and used C++ binding.